### PR TITLE
LPS-152234 Deprecate `submitForm` and replace usages

### PR DESCRIPTION
**Pull request by kresimir-coko**
The initial PR caused some regressions but I think I have fixed them now. The problem was with the `events.js` changes, where I relied on the `form` to be a regular HTML element, not a YUI element.

After consulting with Meg, she gave me a bunch of additional CI checks to run to make sure the teams that reported the regressions are covered.

### CODE REVIEW

**Comment by kresimir-coko** [comment]
ci:test:data-engine

**Comment by kresimir-coko** [comment]
ci:test:acceptance-functional

**Comment by kresimir-coko** [comment]
ci:test:blogs

**Comment by kresimir-coko** [comment]
ci:test:forms-functional

**Comment by kresimir-coko** [comment]
ci:test:commerce-functional

File: modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/custom/form/FormView.es.js [hunk 1]
```diff
@@ -81,7 +81,11 @@ const useFormSubmit = ({apiRef, containerRef}) => {
 							}
 
 							if (submittable) {
-								Liferay.Util.submitForm(event.target);
+								const form = event.target;
+
+								if (form.nodeName === 'FORM') {
```
Comment (bryceosterhaus) [inline modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/custom/form/FormView.es.js:86]: How come we are adding these extra checks? It seems like the Util version simply just calls `.submit()` without checking anything. I think we can do the same

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/events.js [hunk 2]
```diff
@@ -175,7 +175,9 @@
 
 				form.attr('action', action);
 
-				Util.submitForm(form);
+				var formNode = form._node || form;
```
Comment (bryceosterhaus) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/events.js:178]: Same as above, I think we can remove this check and only have `from.submit()` to replace the util

**Review by bryceosterhaus — CHANGES_REQUESTED** [review]

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/events.js (hunk 2 above)
Comment (kresimir-coko) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/events.js:178]: Lack of this check is what caused all those regressions a couple weeks ago. @bryceosterhaus

File: modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/custom/form/FormView.es.js (hunk 1 above)
Comment (kresimir-coko) [reply modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/custom/form/FormView.es.js:86]: Previously when I just had `event.target.submit()` those regressions were failing a couple weeks ago. What happens when `event.target` isn't a `form` element? I think that AUI handles it under the hood whereas JS doesn't.

**Comment by kresimir-coko** [comment]
I'll double check your comments @bryceosterhaus and confirm if we really do 100% require these checks.

File: modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/custom/form/FormView.es.js (hunk 1 above)
Comment (bryceosterhaus) [reply modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/custom/form/FormView.es.js:86]: Hmm, but then the question is why would the old implementation work? The old implementation with `Util.submitForm` was directly calling `.submit()` without any checks.

```
submitForm(form) {
	form.submit();
},
```


If this code works, then we shouldn't need the custom checks, unless something else is going on

**Comment by kresimir-coko** [comment]
ci:test:sf

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by kresimir-coko** [comment]
ci:forward

**Pull request closed without merging**